src/Microsoft.OData.Edm/Vocabularies/Annotations/EdmTermExtensions.cs

```diff
@@ -0,0 +1,446 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="EdmTermExtensions.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using Microsoft.OData.Edm.Csdl;
+using System.Collections.Generic;
+using System.Linq;
+using System.IO;
+using System.Text;
+using Microsoft.OData.Edm.Csdl.Parsing.Ast;
+
+namespace Microsoft.OData.Edm.Vocabularies
+{
+    internal static class EdmTermExtensions
+    {
+        /// <summary>
+        /// Parses a <paramref name="defaultValue"/> into an IEdmExpression value of the correct EDM type.

```

Comment:
>Parses a <paramref name="defaultValue"/> into an IEdmExpression value of the correct EDM type. [](http://example.com/codeflow?start=12&length=94)

More generically, this method Parses a *string value* into an IEdmExpression value of the correct EDM type. takes a string value and generates an IEdmExpression.  As you are currently using it the string represents a default value, but perhaps it could be used other places to translate some other type of string to an expreession.  Perhaps name it more generically; i.e., just BuildEdmExpression(IEdmTypeReference typeReference, string value).

Reply:
I think I called it "BuildDefaultEdmExpression" to distinguish between that and "BuildDefaultCsdlExpression" since I need a different number of parameters for the different constant expression types they return.

Reply:
maybe BuildEdmExpression and BuildCsdlExpression?

Reply:
Ah, I see what you're getting at!

Reply:
Done ✅